packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,258 @@
+    });
+  }
+
+  /**
+   * This method is called when the custom element is disconnected from the document's DOM.
+   */
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+  }
+
+  /**
+   * The dialog element.
+   * @public
+   *
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modalType
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @remarks
+   * HTML Attribute: type
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue start
+   */
+  @attr
+  public position?: DrawerPosition;
+
+  /**
+   * Sets the size of the drawer (small/medium/large).
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   * @defaultValue medium
+   */
+  @attr({ attribute: 'size' })
+  public size?: DrawerSize;
+
+  /**
+   * Sets the open state of the drawer
+   * @public
+   * @remarks
+   * HTML Attribute: open
+   * @defaultValue false
+   */
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Indicates whether the drawer is currently closing.
+   */
+  private closing: boolean = false;
+
+  /**
+   * Emits a 'cancel' event.
+   * @internal
+   */
+  public cancel(): void {
+    this.$emit('cancel');
+  }
+
+  /**
+   * Handles changes to the `open` property.
+   * @public
+   */
+  public openChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected) {
+      if (newValue) {
+        this.show();
+      } else {
+        this.hide();
+      }
+    }
+    this.$emit('onOpenChange', { open: this.open });
+  }
+
+  /**
+   * @public
+   * Method called when the 'modalType' attribute changes
+   */
+  public typeChanged(oldValue: DrawerType, newValue: DrawerType): void {
+    if (newValue !== oldValue) {
+      if (newValue == DrawerType.inline) {
+        this.modalType = DrawerModalType.nonModal;
+      }
+    }
+  }
+
+  /**
+   * Sets the component state based on the `open` property.
+   * @public
+   */
+  public setComponent(): void {
+    if (this.open) {
+      this.show();
+    }
+  }
```

Comment:
IMO we should defer to the dialog's native elements and use and respond to those where appropriate.

Reply:
This was an attempt to ensure that the internal dialog state aligns accurately with the open state of the <fluent-drawer> when the component is first connected to the DOM. By synchronizing the internal <dialog> with the external open attribute at connection, I was aiming for the custom component to reflect the expected native behavior, even though the initial native triggering of the dialog’s display state (show/showModal) is managed by the custom code.